app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataController.kt

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.content.Context
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.withContext
+import mozilla.components.concept.engine.Engine
+import mozilla.components.feature.tab.collections.TabCollection
+import org.mozilla.fenix.ext.components
+import kotlin.coroutines.CoroutineContext
+
+interface DeleteBrowsingDataController {
+    suspend fun deleteTabs()
+    suspend fun deleteBrowsingData()
+    suspend fun deleteCollections(collections: List<TabCollection>)
+    suspend fun deleteCookies()
+    suspend fun deleteCachedFiles()
+    suspend fun deleteSitePermissions()
+}
+
+class DefaultDeleteBrowsingDataController(
+    val context: Context,
+    val coroutineContext: CoroutineContext = Dispatchers.Main
+) : DeleteBrowsingDataController {
+    override suspend fun deleteTabs() {
+        withContext(coroutineContext) {
+            context.components.useCases.tabsUseCases.removeAllTabs.invoke()
+        }
+    }
+
+    override suspend fun deleteBrowsingData() {
```

Comment:
My understanding was that we wanted to update this to only delete history here and separate out the other categories. If we have this one clear `Engine.BrowsingData.all()` then the categories are kind of obsolete

Reply:
But we should probably maintain the old behavior in an if/else via the feature flag